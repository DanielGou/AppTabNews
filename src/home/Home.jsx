import { useState, useEffect } from 'react';
import { StyleSheet, FlatList, ActivityIndicator, View } from 'react-native';
import { footer } from './footer';
import homeService from './homeService'

import LinkPost from './LinkPost'

export default function Home(){
    let homeApi = new homeService()

    const [ posts, setPosts ] = useState([])
    const [ page, setPage ] = useState(1)
    const [ isLoading, setLoading ] = useState(true)
  
    useEffect(()=>{
        loadListPosts()
    },[])

    async function loadListPosts(){
        let listPosts = (await homeApi.getListPosts(page)).data
        if(posts.length > 0){
            setPosts([...posts , ...listPosts])
        }else{
            setPosts(listPosts)
        }
        setPage(page+1)
        setLoading(false)
    }

    const renderFooter = footer(isLoading)

    if(isLoading){
        return(
            <View style={styles.loading}>
                <ActivityIndicator size="large"/>
            </View>
        )
    }else{
        return(
            <FlatList 
                style={styles.container}
                data={posts}
                renderItem={(post)=><LinkPost post={post}/>}
                keyExtractor={item => item.id}    
                onEndReached={loadListPosts}
                onEndReachedThreshold={0.1}
                ListFooterComponent={renderFooter}
            />
        )
    }
}

export const styles = StyleSheet.create({
    container: {
      backgroundColor: '#fff',
    },
    loading:{
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
    },
});


