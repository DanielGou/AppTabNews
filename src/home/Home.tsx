import React, { useState, useEffect } from 'react';
import { StyleSheet, FlatList, ActivityIndicator, View } from 'react-native';
import HomeService from './HomeService'

import LinkPost from './components/LinkPost'
import {Footer} from './components/Footer';

export default function Home() {
    let homeService = new HomeService()

    const [posts, setPosts] = useState<any>([])
    const [page, setPage] = useState(1)
    const [isLoading, setLoading] = useState<any>(true)

    useEffect(() => {
        loadListPosts()
    }, [])

    async function loadListPosts() {
        let listPosts = (await homeService.getListPosts(page))?.data

        if (posts.length > 0) {
            setPosts([...posts, ...listPosts])
        } else {
            setPosts(listPosts)
        }
        setPage(page + 1)
        setLoading(false)
    }

    if (isLoading) {
        return (
            <View style={styles.loading}>
                <ActivityIndicator size="large" />
            </View>
        )
    } else {
        return (
            <FlatList
                style={styles.container}
                data={posts}
                renderItem={(post) => <LinkPost post={post} />}
                keyExtractor={item => item.id}
                onEndReached={loadListPosts}
                onEndReachedThreshold={0.1}
                ListFooterComponent={Footer(isLoading)}
            />
        )
    }
}

export const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
    },
    loading: {
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
    },
});


