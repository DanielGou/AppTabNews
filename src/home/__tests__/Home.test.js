import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';

const tree = renderer.create(<Home />);

test('snapshot App', ()=>{
  expect(tree).toMatchSnapshot();
})
