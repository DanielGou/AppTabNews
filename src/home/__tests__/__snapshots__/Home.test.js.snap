// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;
